lib/elastic_record/relation.rb

```diff
@@ -31,12 +31,6 @@ def becomes(klass)
       became
     end
 
-    def metamorphose(new_klass)
-      result = self.class.new(new_klass, values: values.dup)
-      result.safe! if safe?
-      result
-    end
```

Comment:
I think a `metamorphose` method is still helpful, even if it's now just doing L35.

Reply:
What do you think about overriding `initialize_dup`? [Article here](https://blog.appsignal.com/2019/02/26/diving-into-dup-and-clone.html).

Reply:
How would that look? Right now we're doing something like:

```ruby
elastic_relation.metamorphose(Contact)
```

It's nice to be able to pass the class it's morphing into.

Reply:
oh, give `ElasticRecord::Relation#becomes` a try.

https://github.com/data-axle/elastic_record/blob/b7ea102f288ff1e35a346c1c904932515a95e8b3/lib/elastic_record/relation.rb#L27-L32

Reply:
Yeah, it's the `@klass` value that we want to set. (Which confusingly here is different from the `klass` value)

Reply:
Just coming back to this after a bit, I added `Relation#metamorphose` back.